ipatests/test_integration/test_installation.py

```diff
@@ -1119,6 +1129,110 @@ def run_request(url, expected_stdout=None, expected_stderr=None):
+                e.sendline(self.master.config.admin_password)
+                e.expect_exact('Password (confirm): ')
+                e.sendline(self.master.config.admin_password)
+                e.expect_exact('Do you want to configure chrony with '
+                               'NTP server or pool address? [no]: ')
+                e.sendline('no')
+                e.expect_exact('Continue to configure the system '
+                               'with these values? [no]: ')
+                e.sendline('yes')
+                e.expect_exit(ignore_remaining_output=True, timeout=720)
+            hostname = self.master.run_command([
+                'hostname', '-f']).stdout_text.strip()
+            assert hostname == new_hostname
+        finally:
+            # no need to restore the hostname as the installer
+            # does it during uninstallation
+            self.master.put_file_contents(paths.HOSTS, hosts)
+
+    def test_ad_subpackage_dependency(self, server_cleanup):
+        """
+        Test if the installer is not dependant on trust-ad package and
+        succeeds even when AD subpackage is not installed
+
+        https://pagure.io/freeipa/issue/4011
+        """
+        package_name = '*ipa-server-trust-ad'
+        reinstall = False
+        if tasks.is_package_installed(self.master, package_name):
+            tasks.uninstall_packages(self.master, [package_name])
+            reinstall = True
+        try:
+            tasks.install_master(self.master)
+        finally:
+            if reinstall:
+                tasks.install_packages(self.master, [package_name])
+
+    def test_backup_of_cs_cfg_is_created(self, server_cleanup):
+        """
+        Test that the installer backs up CS.cfg configuration before it's
+        being modified.
+
+        https://pagure.io/freeipa/issue/4166
+        """
+        bcp_location = paths.CA_CS_CFG_PATH + '.ipabkp'
+        original_cfg_content = None
+        if self.master.transport.file_exists(paths.CA_CS_CFG_PATH):
+            original_cfg_content = self.master.get_file_contents(
+                paths.CA_CS_CFG_PATH, encoding='utf-8')
+
+        time_before_install = self.master.run_command(
+            ['date', '+%s']).stdout_text.strip()
+        tasks.install_master(self.master)
+        ipaserver_install_log = self.master.get_file_contents(
+            paths.IPASERVER_INSTALL_LOG, encoding='utf-8')
+        assert 'backing up CS.cfg' in ipaserver_install_log
+        assert self.master.transport.file_exists(bcp_location)
+
+        cfg_mod_time = self.master.run_command(
+            ['stat', '-c', '%Y', paths.CA_CS_CFG_PATH]).stdout_text.strip()
+        bcp_mod_time = self.master.run_command(
+            ['stat', '-c', '%Y', bcp_location]).stdout_text.strip()
+
+        # check if the backup file was modified/created during the installation
+        # and before the original file was modified
+        assert time_before_install <= bcp_mod_time <= cfg_mod_time
```

Comment:
You are comparing numbers as strings. This is generally a bad idea, please convert them to numbers. This will  verify that the output of `date` and `stat` is in expected format.